### PR TITLE
Fix golint warnings in stat package

### PR DESCRIPTION
**Pull request by vthiery**
Fixing the warnings that `golint` highlights in the `stat` package

### CODE REVIEW

File: stat/distmv/studentst.go [hunk 1]
```diff
@@ -226,17 +226,17 @@ func findUnob(observed []int, dim int) (unobserved []int) {
 //  covariance(i, j) = E[(x_i - E[x_i])(x_j - E[x_j])]
 // If the input matrix is nil a new matrix is allocated, otherwise the result
 // is stored in-place into the input.
-func (st *StudentsT) CovarianceMatrix(s *mat.SymDense) *mat.SymDense {
-	if s == nil {
-		s = mat.NewSymDense(st.dim, nil)
+func (s *StudentsT) CovarianceMatrix(m *mat.SymDense) *mat.SymDense {
```
[the remark this answers is not in the document]
Comment (vthiery) [reply stat/distmv/studentst.go]: The change of `st` to `s` is suggested by `golint` and makes the name consistent with the rest of the methods. The input matrix (here named `m`) is called `s` for the normal distribution and `cov` for the Dirichlet distribution. Wouldn't it make sense to name it consistently for these 3 distributions?

[the remark this answers is not in the document]
Comment (kortschak) [reply stat/distmv/studentst.go]: `golint` makes style suggestions, not correctness warnings and so it completely ignorable. Conventionally we use `m`, `s` and `t` (etc) for general, symmetric, and triangular matrices respectively (not a hard restrictions - see below). The use of `st` for the receiver reflects what the receiver type is nicely.

In this case, I think the `s` parameter should probably actually be `dst` to more clearly indicate that it is actually a destination value.

[the remark this answers is not in the document]
Comment (vthiery) [reply stat/distmv/studentst.go]: Got it! I reverted this change

**Comment by kortschak** [comment]
Thank you

**Pull request merged**